drivers/gles2/shaders/scene.glsl

```diff
@@ -433,17 +433,44 @@ void main() {
 #else
 	// look up transform from the "pose texture"
 	{
-		for (int i = 0; i < 4; i++) {
-			ivec2 tex_ofs = ivec2(int(bone_ids[i]) * 3, 0);
+		ivec4 bone_indicesi = ivec4(bone_ids); // cast to signed int
 
-			highp mat4 b = mat4(
-					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(0, 0)),
+		ivec2 tex_ofs = ivec2(bone_indicesi.x * 3, 0);
+		bone_transform = mat4(
+					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs),
 					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(1, 0)),
 					texel2DFetch(bone_transforms, skeleton_texture_size, tex_ofs + ivec2(2, 0)),
-					vec4(0.0, 0.0, 0.0, 1.0));
+					vec4(0.0, 0.0, 0.0, 1.0)) *
```

Comment:
Is this the same as `vec4(0.0, 0.0, 0.0, bone_weights.x)`? (or maybe you were thinking the way you have done would be cheaper?)